### PR TITLE
xdg updates + autostart + lockfile

### PR DESCRIPTION
**Pull request by munix9**
I have summarized the various changes here:

- Use of pyxdg
- Extension of xdg.py (autostart)
- control autostart in tray
- check if os-ui is already running

See also https://github.com/evilsocket/opensnitch/issues/950

### CODE REVIEW

**Comment by gustavo-iniguez-goya** [comment]
This looks really good to me @munix9 , removing the pkill was a must.

Some general notes:

I see that pyxdg is packaged as python3-xdg on debian and ubuntu, and as python3-pyxdg/python36-pyxdg on fedora and opensuse.
Only available since ubuntu >= 20.04, so we would be incompatible with ubuntu 18 and 16 (and the linux mint/mx equivalents). Just a note, nothing really important, but worth mentioning.

File: ui/bin/opensnitch-ui [hunk 1]
```diff
@@ -40,11 +40,13 @@ if dist_path not in sys.path:
 from opensnitch.service import UIService
 from opensnitch.config import Config
 from opensnitch.utils import Themes, Utils, Versions
+from opensnitch.utils.xdg import xdg_runtime_dir
 import opensnitch.ui_pb2
 from opensnitch.ui_pb2_grpc import add_UIServicer_to_server
 
 def on_exit():
     server.stop(0)
+    lockfile.unlock()
```
Comment (gustavo-iniguez-goya) [inline ui/bin/opensnitch-ui:49]: on my system this complains about `lockfile is not defined`
a missing import?

**Comment by munix9** [comment]
> This looks really good to me @munix9 , removing the pkill was a must.
> 
> Some general notes:
> 
> I see that pyxdg is packaged as python3-xdg on debian and ubuntu, and as python3-pyxdg/python36-pyxdg on fedora and opensuse. Only available since ubuntu >= 20.04, so we would be incompatible with ubuntu 18 and 16 (and the linux mint/mx equivalents). Just a note, nothing really important, but worth mentioning.

This is not nice. Maybe I can integrate the needed methods of pyxdg directly into xdg.py to avoid the dependency.
I'll have a look at it.

File: ui/bin/opensnitch-ui (hunk 1 above)
Comment (munix9) [reply ui/bin/opensnitch-ui:49]: That's strange, it works for me without any problem.

`lockfile` is defined in line 99 https://github.com/evilsocket/opensnitch/blob/d003cc162c56db1d69bd62265ef3b5216dac4a05/ui/bin/opensnitch-ui#L99, directly after `app`.

What versions (python, qt) are you using?

Comment (gustavo-iniguez-goya) [reply ui/bin/opensnitch-ui:49]: nah, nevermind, my local branch of the PR was not updated, sorry.

**Comment by munix9** [comment]
@gustavo-iniguez-goya pyxdg dependency is removed.

**Comment by gustavo-iniguez-goya** [comment]
@munix9 , did you see this error? it occurs when the `~/.config/autostart/opensnitch_ui.desktop` already exists. We'll hit this error when users coming from v1.5.x install the next version.

```python
  File "/usr/lib/python3/dist-packages/opensnitch/service.py", line 202, in _on_switch_autostart
    self._autostart.enable(self._menu_autostart.isChecked())
  File "/usr/lib/python3/dist-packages/opensnitch/utils/xdg.py", line 82, in enable
    shutil.copyfile(self.systemAutostart, self.userAutostart)
  File "/usr/lib/python3.8/shutil.py", line 244, in copyfile
    raise SameFileError("{!r} and {!r} are the same file".format(src, dst))
shutil.SameFileError: '/etc/xdg/autostart/opensnitch_ui.desktop' and '/home/ga/.config/autostart/opensnitch_ui.desktop' are the same file
```

**Comment by gustavo-iniguez-goya** [comment]
thank you @munix9 !

**Pull request merged**